Add Ruby assignment 8 tdd

shirishwebonise opened the pull request:
-  used rspec for test specifictions

shirishparsekar committed:
Add ruby assignment - TDD, EOD

shirishparsekar committed:
Remove roman-numeral gem

shirishparsekar committed:
Remove roman-numeral gem usage and impletement conversion algorithm

shirishparsekar committed:
Add tests for single number conversions

tejaswinic-webonise commented on Ruby-Assignment-8-TDD/roman-number-coding/lib/roman_code_evaluator.rb:
```diff
@@ -0,0 +1,38 @@
+
+class RomanCodeEvaluator
+  @@CODE = { coin: 'I', dollar: 'V', yen: 'X', rupee: 'L' }
```
try not to use class var

tejaswinic-webonise commented on Ruby-Assignment-8-TDD/roman-number-coding/spec/roman_number_spec.rb:
```diff
@@ -0,0 +1,64 @@
+      expect(roman_number.to_decimal).to eq 0
+    end
+  end
+end
```
try to add negative test cases too.

tejaswinic-webonise commented on Ruby-Assignment-8-TDD/roman-number-coding/lib/roman_code_evaluator.rb:
```diff
@@ -0,0 +1,38 @@
+
+  def initialize
+    @multiplicatives = Hash.new
+    eval_multiplicative("coin coin Silver 34")
```
Actually its none of the concern of TDD assignment , but do not call method from initialize, You can call it by the instance of the same class.

tejaswinic-webonise commented on Ruby-Assignment-8-TDD/roman-number-coding/lib/roman_number.rb:
```diff
@@ -0,0 +1,33 @@
+  def to_decimal
+    converted_num = 0
+    last_decimal = 0
+    @number_str.split('').reverse.each do |element|
```
Try: this loop should reside in another method, and to decimal would give only decimal value for given input.

shirishparsekar committed:
Separate out code hash into its own class

shirishparsekar committed:
Write specifications for RomanCode class

shirishparsekar committed:
Refactor to clean the code
 - rename mappings to MAPPINGS to make it constant
 - raise error if input is not in proper roman form
 - rename method to_decimal to decimal_value
 - extract method from implementation of decimal_value

shirishparsekar committed:
Refactor

shirishparsekar committed:
Add specification for negative flow
